tests/testsuite.src/data_binary.at

```diff
@@ -2265,3 +2265,452 @@ AT_CHECK([$COMPILE prog.cob], [0], [], [])
 AT_CHECK([$COBCRUN_DIRECT ./prog], [0], [], [])
 
 AT_CLEANUP
+
+
+AT_SETUP([Fixed BINARY size])
+AT_KEYWORDS([binary])
+
+AT_DATA([prog.cob], [
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. prog.
+
+       DATA DIVISION.
+       WORKING-STORAGE SECTION.
+       01 WS-BCS    BINARY-CHAR   SIGNED.
+       01 WS-BCU    BINARY-CHAR   UNSIGNED.
+       01 WS-BSS    BINARY-SHORT  SIGNED.
+       01 WS-BSU    BINARY-SHORT  UNSIGNED.
+       01 WS-BLS    BINARY-LONG   SIGNED.
+       01 WS-BLU    BINARY-LONG   UNSIGNED.
+       01 WS-BDS    BINARY-DOUBLE SIGNED.
+       01 WS-BDU    BINARY-DOUBLE UNSIGNED.
+
+       PROCEDURE DIVISION.
+       MAIN.
+           IF BYTE-LENGTH(WS-BCS) <> 1
+               DISPLAY "BAD BINARY-CHAR SIGNED".
+           IF BYTE-LENGTH(WS-BCU) <> 1
+               DISPLAY "BAD BINARY-CHAR UNSIGNED".
+           IF BYTE-LENGTH(WS-BSS) <> 2
+               DISPLAY "BAD BINARY-SHORT SIGNED".
+           IF BYTE-LENGTH(WS-BSU) <> 2
+               DISPLAY "BAD BINARY-SHORT UNSIGNED".
+           IF BYTE-LENGTH(WS-BLS) <> 4
+               DISPLAY "BAD BINARY-LONG SIGNED".
+           IF BYTE-LENGTH(WS-BLU) <> 4
+               DISPLAY "BAD BINARY-LONG UNSIGNED".
+           IF BYTE-LENGTH(WS-BDS) <> 8
+               DISPLAY "BAD BINARY-DOUBLE SIGNED".
+           IF BYTE-LENGTH(WS-BDU) <> 8
+               DISPLAY "BAD BINARY-DOUBLE UNSIGNED".
+           STOP RUN.
+])
+
+AT_CHECK([$COMPILE -fintrinsics=ALL -Wno-constant-expression -fbinary-size=1-2-4-8 prog.cob], [0], [], [])
```

Comment:
you may opt to use `-fintrinsics=BYTE-LENGTH` instead or, likely better, include `REPOSITORY. FUNCTION BYTE-LENGTH INTRINSIC.` to the code allowing to test the program with other compilers as well (same below)